src/legend_data_monitor/plotting.py

```diff
@@ -450,48 +449,64 @@ def plot_per_string(data_analysis, plot_info, pdf):
 
     data_analysis.data = data_analysis.data.sort_values(["location", "label"])
     # new subplot for each string
-    ax_idx = 0
     for location, data_location in data_analysis.data.groupby("location"):
-        utils.logger.debug(f"... {plot_info['locname']} {location}")
+        # if string number specified, this function is being called by external code
+        # and is not invoked by make_subsystem_plots()
+        if string:
+            if string[0] != location:
+                continue
+            max_ch_per_string = (
+                data_analysis.data.groupby("location")["position"].nunique().max()
+            )
+            global COLORS
+            COLORS = color_palette("hls", max_ch_per_string).as_hex()
+
+        # otherwise just go on with standard code
+        else:
+            utils.logger.debug(f"... {plot_info['locname']} {location}")
+        # create one different figure per string
+        fig, axes = plt.subplots(figsize=(15, 5))
 
         # new color for each channel
         col_idx = 0
         labels = []
         for label, data_channel in data_location.groupby("label"):
             ch_dict = plot_style(
-                data_channel, fig, axes[ax_idx], plot_info, COLORS[col_idx]
+                data_channel, fig, axes, plot_info, color=COLORS[col_idx]
             )
             labels.append(label)
             col_idx += 1
-
             channel = ((label.split("-")[1]).split("ch")[-1]).lstrip("0")
             if channel not in par_dict.keys():
                 par_dict[channel] = ch_dict
 
         # add grid
-        axes[ax_idx].grid("major", linestyle="--")
+        axes.grid("major", linestyle="--")
         # beautification
-        axes[ax_idx].set_title(f"{plot_info['locname']} {location}")
-        axes[ax_idx].set_ylabel("")
-        axes[ax_idx].legend(labels=labels, loc="center left", bbox_to_anchor=(1, 0.5))
+        axes.set_title(f"{plot_info['locname']} {location}")
+        axes.set_ylabel("")
+        axes.legend(labels=labels, loc="center left", bbox_to_anchor=(1, 0.5))
 
         # plot the position of the two K lines
-        if plot_info["cuts"] == "K lines":
-            axes[ax_idx].axhline(y=1460.822, color="gray", linestyle="--")
-            axes[ax_idx].axhline(y=1524.6, color="gray", linestyle="--")
+        if plot_info["title"] == "K lines":
```

Comment:
safer to leave `plot_info["cuts"]` ("title" is decided by the user, so it could also be "potassium lines" or "K events" and at that point you don't enter into this `if`, even if you want to).
The `cuts` option for selecting K events in the near future will ALWAYS be needed (at the moment, if someone does not put it, the code dosn't crash but it will give an empty plot - in the future, we will raise an error if not present)

Reply:
Oh this was not intentional, I'll fix this